src/ref.js

```diff
@@ -9,10 +10,16 @@ function isNewRef(prop) {
   return ownProperties.length === 1 && ownProperties[0] === 'current';
 }
 
+function isCallbackRef(prop) {
+  return typeof prop === 'function'
+    && !Object.prototype.isPrototypeOf.call(Component, prop)
```

Comment:
The linter does not want `Component.isPrototypeOf(prop)`.

Reply:
we should also bring in `PureComponent` and add:
```suggestion
    && !Object.prototype.isPrototypeOf.call(Component, prop)
    && (!PureComponent || !Object.prototype.isPrototypeOf.call(PureComponent, prop))
```

Reply:
Yeah, you're right. I thought PureComponent extended Component and we wouldn't need this. Not the case, however.

Reply:
So, what's interesting is that the tests for handling Component and PureComponent pass even if we remove both `isPrototypeOf` checks.

Turns out that [Components and PureComponents have more than 1 argument to their constructors](https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/react/src/ReactBaseClasses.js#L131), and therefore don't pass the arity check.

We could simplify this a bit and remove the checks against Component/PureComponent. I think that makes sense, but there's also an argument to be made for explicitly checking against them. @ljharb do you have thoughts on that?

Reply:
That's not necessarily true; `class extends React.Component { constructor(props) { super(...props); } }` will have a length of 1 but shouldn't pass the check.

Reply:
Ah, of course. This should be good to go, then